protocol/bc/asset.go

```diff
@@ -22,13 +22,17 @@ func (a *AssetID) Scan(b interface{}) error     { return (*Hash)(a).Scan(b) }
 
 // ComputeAssetID computes the asset ID of the asset defined by
 // the given issuance program and initial block hash.
-func ComputeAssetID(issuanceProgram []byte, initialHash [32]byte, vmVersion uint64) (assetID AssetID) {
+func ComputeAssetID(issuanceProgram []byte, initialHash [32]byte, vmVersion uint64, assetDefinitionHash Hash) (assetID AssetID) {
 	h := sha3pool.Get256()
 	defer sha3pool.Put256(h)
 	h.Write(initialHash[:])
-	blockchain.WriteVarint63(h, assetVersion)
 	blockchain.WriteVarint63(h, vmVersion)
 	blockchain.WriteVarstr31(h, issuanceProgram) // TODO(bobg): check and return error
+	if assetDefinitionHash == EmptyStringHash {
```

Comment:
Why check for this? It seems like it would be easier to just hash the asset definition hash regardless. Presumably there's a drawback to that but I don't understand what it is.